jenkins/shared-library/vars/fileHelper.groovy

```diff
@@ -0,0 +1,18 @@
+import java.nio.file.Paths
+
+String getTempPathInWorkspace(String filepath) {
+    return Paths.get("${WORKSPACE_TMP}", filepath)
+}
+
+String copyToTempFile(String data, String destinationFilepath) {
+    return copyToLocalFile(data, getTempPathInWorkspace(destinationFilepath))
+}
+
+String copyToLocalFile(String data, String destinationFilepath) {
+    String parentPath = Paths.get(destinationFilepath).getParent()
+    dir (parentPath) {
```

Comment:
why do you need to execute in parentPath? aren't you providing full filepath to writeFile?